controllers/mealsController.js

```diff
@@ -1,29 +1,79 @@
 const Meal = require('../models').Meal
 const User = require('../models').User
+const Food = require('../models').Food
+const MealFood = require('../models').MealFood
 const pry = require('pryjs')
 
 const create = (req, res) => {
-  User.findOne({ where: { api_key: req.body.api_key } })
+  findUser(req.body.api_key)
     .then(user => {
-      let meal = req.body.meal
-      meal = meal.toLowerCase()
-      meal = meal[0].toUpperCase() + meal.substring(1)
-      Meal.findOrCreate({ where: {
-        meal_name: meal,
-        date: req.body.date
-      } })
-      .then(meal => {
-        res.setHeader("Content-Type", "application/json");
-        res.status(201).send(JSON.stringify({"Message": `${meal[0].meal_name} has been added to meals for ${meal[0].date.toLocaleDateString('en-US')}`})) })
-      .catch(error => {
-        res.setHeader("Content-Type", "application/json");
-        res.status(400).send(JSON.stringify("Well that didn't work.")) })
+      const meal = sanitizeEntry(req.body.meal)
+      return createMeal(meal, req.body.date, res)
     .catch(error => {
       res.setHeader("Content-Type", "application/json");
       res.status(401).send(JSON.stringify("Invalid credentials")) })
     });
 }
 
+const createMeal = (meal, date, res) => {
+  Meal.findOrCreate({ where: {
+    meal_name: meal,
+    date: date
+  } })
+  .then(meal => {
+    res.setHeader("Content-Type", "application/json");
+    res.status(201).send(JSON.stringify({ "id": `${meal[0].id}`, "Message": `${meal[0].meal_name} has been added to meals for ${meal[0].date.toLocaleDateString('en-US')}` } )) })
+  .catch(error => {
+    res.setHeader("Content-Type", "application/json");
+    res.status(400).send(JSON.stringify("Could not store meal.")) })
+}
+
+const sanitizeEntry = (userEntry) => {
```

Comment:
I didn't put this functionality into updating a food, I wanted to try setting this in it's own file to use it across our routes for POST and PUT.

Reply:
Totally, I would love to do that in one of our early refactors, once we get base functionality down.